Core/Src/speed_calc.c

```diff
@@ -13,31 +13,32 @@
 #include "datalog_handler.h"
 #include "main.h"
 
-//distancia 	= circuferencia dividido pela quantidade de dentes
-//				= 2*pi*raio/quant.dentes (m)
-//tempo 		= valor do timer multiplicado pelo prescaler dividido pela frequencia
-//				= timer*prescaler/frequencia (s)
-//velocidade	= distancia/tempo
+//distancia     = circuferencia dividido pela quantidade de dentes
+//              = 2*pi*raio/quant.dentes (m)
+//tempo         = valor do timer multiplicado pelo prescaler dividido pela frequencia
+//              = timer*prescaler/frequencia (s)
+//velocidade    = distancia/tempo
 //              = (2*pi*raio/quant.dentes)*freq/(presc*timer) (m/s)
 //              = (10*3.6*2*pi*raio/quant.dentes)*freq/(presc*timer) (10*km/s)
 
-speed_pin_e get_speed_pin(uint16_t pin);
+void reset_speed_all();
+void reset_speed_single(speed_message_t message, speed_message_t last_messages[4], uint32_t min_count);
```

Comment:
passa essas coisas por referência, se n toda vez que vc chamar essa função ele vai fazer uma cópia das structs:
```suggestion
void reset_speed_single(speed_message_t* message, speed_message_t* last_messages, uint32_t min_count);
```